### DISCUSSION TITLE
Kiali pod logs always throwing TLS handsshake error log message and audit logs

### DISCUSSION THREAD

**Discussion by Vineeth-varma** [kiali/kiali#4901] [Q&A]
Hi I am running kiali using Kiali Operator, I have two questions here 

1. Why kiali pods are always displaying the below error messages 
```
2022/04/04 05:55:59 http: TLS handshake error from 10.192.40.221:33292: EOF
2022/04/04 05:56:03 http: TLS handshake error from 10.192.40.221:30660: EOF
2022/04/04 05:56:06 http: TLS handshake error from 10.192.40.221:38723: EOF
2022/04/04 05:56:09 http: TLS handshake error from 10.192.40.221:40386: EOF
```

2. In document it mentioned by default audit logs will be captured to pod logs when we run command ``` kubectl logs <podname> -n namespace ``` 
3. What does this audit logs mean, because I have not seen any logs displaying authentication failed messages or passed wrong token like this...,

Thanks in Advance

**Comment by lucasponce** [comment]
Could you describe the environment, Istio version, and the steps you did to install Kiali?

**Comment by Vineeth-varma** [comment]
I took helm charts from the community,and deployed with latest versions.

**Comment by jmazzitelli** [comment]
I'm not sure because you didn't say how you access the Kiali UI (what URL do you point your browser to?) and what the Kiali config is (particularly are you on Kubernetes? and did you install using the operator helm chart or server helm chart? and did you assign identity to the kiali server via the Kiali CR [spec.identity](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.identity)?)

But I think this message shows up if you are trying to get to Kiali over https when the Kiali Server has not been told to start up an HTTP server behind SSL. Again, I can't say for sure because we don't have enough information.

**Comment by Vineeth-varma** [comment]
I have deployed kiali in EKS cluster of amazon. And accessing kiali using amazon Load Balancer, and I have used operator helm chart. I even have assigned identity to the kiali server so accessing kiali via https.

Am I missing some configuration here? @jmazzitelli

**Reply by jmazzitelli** [reply]
Does it look like everything is working? I still have very little details. So, sorry, there isn't much I can go on. But if you are saying the Kiali UI looks good in your browser but yet you are seeing these log messages, to me it seems like SOME client out there is probing your Kiali using a `http` request (rather than https) and the server is logging this fact. I'm pretty sure `http: TLS handshake error from 10.192.40.221:33292: EOF` means somebody is trying to make an `http` request into your Kiali server, but the Kiali Server requires `https`. But that's just a guess.

**Comment by Vineeth-varma** [comment]
yeah I am not facing any issue actually Kiali is working absolutely fine, but dont know why this tls handshake coming all the time. If you need any other details just let me know I will give you @jmazzitelli .

**Reply by jmazzitelli** [reply]
Find out who is making those HTTP requests. Since you say you don't see any issues facing Kiali, then that means your requests aren't causing those errors. I suspect there is someone or something trying to get to the Kiali UI via `http`. You can try to inspect your network and see if you can find out where those requests are coming from. I do not believe there is anything on the Kiali side that can help track this down.

**Reply by nrfox** [reply]
Perhaps the kiali pod's health probe?

**Reply by jmazzitelli** [reply]
IIRC the health probe ~shouldn't be going over that https port~ _(update: they do, but that's not a problem)_. If that were the case, I would think the default behavior on OpenShift would always show that error in the logs (since by default on OpenShift kiali starts up with https).

**Reply by jmazzitelli** [reply]
I'm checking this to make sure its not the health probes...

**Reply by jmazzitelli** [reply]
Ah.. no, it isn't the health probes - we set the scheme appropriately: https://github.com/kiali/kiali-operator/blob/f6f8adad50d167b722ca504b315a8ced030ffc30/roles/default/kiali-deploy/templates/openshift/deployment.yaml#L74-L81

**Reply by jmazzitelli** [reply]
And I just installed Kiali with https - I see this in the logs:
```
2022-04-06T14:34:57Z INF Server endpoint will require https
```
And I do not see that error message.  Also, note the OP is showing those error messages every 3 seconds. Our health checks are[ every 30 seconds](https://github.com/kiali/kiali-operator/blob/f6f8adad50d167b722ca504b315a8ced030ffc30/roles/default/kiali-deploy/templates/openshift/deployment.yaml#L76-L83). So that wouldn't be it.

**Reply by nrfox** [reply]
Just guessing but that rules it out :)
